sklearn/tree/export.py

```diff
@@ -778,3 +780,179 @@ def export_graphviz(decision_tree, out_file=None, max_depth=None,
+        The decision tree estimator to be exported.
+        It can be an instance of
+        DecisionTreeClassifier or DecisionTreeRegressor.
+
+    feature_names : list, optional (default=None)
+        A list of length n_features containing the feature names.
+        If None generic names will be used ("feature_0", "feature_1", ...).
+
+    max_depth : int, optional (default=10)
+        Only the first max_depth levels of the tree are exported.
+        Truncated branches will be marked with "...".
+
+    spacing : int, optional (default=3)
+        Number of spaces between edges. The higher it is, the wider the result.
+
+    decimals : int, optional (default=2)
+        Number of decimal digits to display.
+
+    show_weights : bool, optional (default=False)
+        If true the classification weights will not be exported
+        on each leaf. The classification weights are the number
+        of samples each class.
+
+    Returns
+    -------
+    report : string
+        Text summary of all the rules in the decision tree.
+
+    Examples
+    -------
+
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.tree import DecisionTreeClassifier
+    >>> from sklearn.tree.export import export_text
+    >>> iris = load_iris()
+    >>> X = iris['data']
+    >>> y = iris['target']
+    >>> decision_tree = DecisionTreeClassifier(random_state=0, max_depth=2)
+    >>> decision_tree = decision_tree.fit(X, y)
+    >>> r = export_text(decision_tree, feature_names=iris['feature_names'])
+    >>> print(r)
+    |--- petal width (cm) <= 0.80
+    |   |--- class: 0
+    |--- petal width (cm) >  0.80
+    |   |--- petal width (cm) <= 1.75
+    |   |   |--- class: 1
+    |   |--- petal width (cm) >  1.75
+    |   |   |--- class: 2
+    ...
+    """
+    check_is_fitted(decision_tree, 'tree_')
+    tree_ = decision_tree.tree_
+    class_names = decision_tree.classes_
+    right_child_fmt = "{} {} <= {}\n"
+    left_child_fmt = "{} {} >  {}\n"
+    truncation_fmt = "{} {}\n"
+
+    if max_depth < 0:
+        raise ValueError("max_depth bust be >= 0, given %d" % max_depth)
+
+    if (feature_names is not None and
+            len(feature_names) != tree_.n_features):
+        raise ValueError("feature_names must contain "
+                         "%d elements, got %d" % (tree_.n_features,
+                                                  len(feature_names)))
+
+    if spacing <= 0:
+        raise ValueError("spacing must be > 0, given %d" % spacing)
+
+    if decimals < 0:
+        raise ValueError("decimals must be >= 0, given %d" % decimals)
+
+    if isinstance(decision_tree, DecisionTreeClassifier):
+        value_fmt = "{}{} weights: {}\n"
+        if not show_weights:
+            value_fmt = "{}{}{}\n"
+    else:
+        value_fmt = "{}{} value: {}\n"
+
+    if feature_names:
+        feature_names_ = [feature_names[i] for i in tree_.feature]
+    else:
+        feature_names_ = ["feature_{}".format(i) for i in tree_.feature]
```

Comment:
use ' ' instead of '_' I think

Reply:
Or use `x[i]`??

Reply:
I feel that `feature_27` is more readable than `feature27` and that `x[27]` may be confusing as you may actually have a variable called `x` in your code.

Reply:
Sorry, I meant using a space rather than an underscore, not dropping the underscore